src/Compilers/CSharp/Portable/Binder/Binder_AnonymousTypes.cs

```diff
@@ -223,7 +223,7 @@ private TypeSymbol GetAnonymousTypeFieldType(BoundExpression expression, CSharpS
                         errorArg = expressionType;
                         expressionType = CreateErrorType(SyntaxFacts.GetText(SyntaxKind.VoidKeyword));
                     }
-                    else if (expressionType.IsUnsafe())
```

Comment:
>IsUnsafe [](http://example.com/codeflow?start=44&length=8)

I'm not sure when we get here with a `void*` but could we also get here with a `void*[]`?

Reply:
We get here with `void*` with a scenario like:
```
    static void* M(void* b)
    {
        var a = new { F = b };
        return a.F;
    }
```
That scenario remains an error (it's a pointer type).

On the other hand, scenarios with `void*[]` are now allowed:
```
var a = new { F = new void*[0] };
```

Reply:
might be worthwhile to have a comment to this effect.